File: JAVA/S1_BASIC/L5_BASICRECURSION/sumN.java

class recursion{
    int sum(int n){
        if(n==0)return 0;
        return n+sum(n-1);
    }
}
public class sumN{
    public static void main(String[]args){
        recursion r=new recursion();
        System.out.print(r.sum(5));
    }
}

File: python/S1_BASIC/L5_BASICRECURSION/sumN.py
class Recursion:
    def __init__(self,n):
        self.n=n
        self.sum=0
    def sumN(self,n):
        if n==0:
            return self.sum
        self.sum+=n
        return self.sumN(n-1)
if __name__=="__main__":
    n=int(input())
    R=Recursion(n)
    print(R.sumN(R.n))
        